richtextfx/src/main/java/org/fxmisc/richtext/StyledTextArea.java

```diff
@@ -1163,30 +1162,30 @@ public void dispose() {
         virtualFlow.dispose();
     }
 
-    public UndoManager createPlainLinearUndoManager(UndoManagerFactory factory) {
-        return model.createPlainLinearUndoManager(factory);
+    public UndoManager createPlainUndoManager(UndoManagerFactory<PlainTextChange> factory, int capacity) {
```

Comment:
`int capacity` could be changed into a wrapper class (much like your `CaretOffsetX` class to force users to examine the javadoc and see that a value of `-1` means unlimtied, `0` zero, and `n` is the size of the capacity:
````java
public ChangeQueueCapacity {
    public final static ChangeQueueCapacity UNLIMITED = new ChangeQueueCapcity(-1);
    public final static ChangeQueueCapacity ZERO = new ChangeQueueCapcity(0);

    private final int capacity;
    public final int getCapacity() { return capacity; }

    public ChangeQueueCapacity(int capacity) {
        this.capacity = capacity;
    }
}
````